utils/cache/filecache/filecache.go

```diff
@@ -0,0 +1,208 @@
+package filecache
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+	"time"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+	"github.com/ARM-software/golang-utils/utils/logs/logrimp"
+	"github.com/ARM-software/golang-utils/utils/parallelisation"
+	"github.com/ARM-software/golang-utils/utils/retry"
+)
+
+var closedErr = commonerrors.New(commonerrors.ErrConflict, "cache is closed")
+
+type Cache struct {
+	entries       iEntryMap
+	entriesLM     iLockMap
+	entryProvider IEntryRetriever
+	fs            filesystem.FS
+	cfg           *FileCacheConfig
+	cancelStore   *parallelisation.CancelFunctionStore
+	closed        atomic.Bool
+}
+
+func (c *Cache) gc(ctx context.Context, _ time.Time) {
+	if err := parallelisation.DetermineContextError(ctx); err != nil {
+		return
+	}
+
+	c.entries.Range(func(key string, entry ICacheEntry) {
+		if entry.IsExpired() && c.entriesLM.TryLock(key) {
+			if err := entry.Delete(ctx); err == nil {
+				c.entries.Delete(key)
+				defer c.entriesLM.Delete(key)
+			}
+			defer c.entriesLM.Unlock(key)
```

Comment:
move that to just after the tryLock, otherwise it won't get run if the if statement isn't entered